src/components/Layout/DataTypes.tsx

```diff
@@ -0,0 +1,283 @@
+import PropTypes from 'prop-types'
+import {
+  BaseValue,
+  DataType,
+  DataTypeMap,
+  Type
+} from '@glif/filecoin-actor-utils'
+
+import { Line, NullishLine } from './Lines'
+import { AddressLink } from '../LabeledText/AddressLink'
+import { FilecoinNumber } from '@glif/filecoin-number'
+
+/**
+ * DataTypeLines
+ * Renders a DataType on multiple lines (depending on the contents)
+ */
+
+export const DataTypeLines = ({
+  label,
+  depth,
+  dataType
+}: DataTypeLinesProps) => {
+  switch (dataType.Type) {
+    case Type.Bool:
+    case Type.String:
+    case Type.Number:
+      return <DataTypeLine label={label} depth={depth} dataType={dataType} />
+
+    case Type.Array:
+      const { Contains } = dataType
+      const value = dataType.Value as Array<BaseValue>
+      return (
+        <>
+          {value.map((v, i) => (
+            <DataTypeLine
+              key={i}
+              label={i === 0 ? label : ''}
+              depth={depth}
+              dataType={{ ...Contains, Value: v }}
+            />
+          ))}
+        </>
+      )
+
+    case Type.Object:
+      const { Name, Children } = dataType
+
+      if (Name === 'Cid') {
+        const cid = Children['/'].Value as string
+        return (
+          <DataTypeLine
+            label={`${label} (CID)`}
+            depth={depth}
+            dataType={{ Name, Type: Type.String, Value: cid }}
+          />
+        )
+      }
+
+      return (
+        <DataTypeMapLines label={label} depth={depth} dataTypeMap={Children} />
+      )
+
+    default:
+      throw new Error(`Unexpected DataType: ${JSON.stringify(dataType)}`)
+  }
+}
+
+interface DataTypeLinesProps {
+  label?: string
+  depth?: number
+  dataType: DataType
+}
+
+DataTypeLines.propTypes = {
+  label: PropTypes.string,
+  depth: PropTypes.number,
+  dataType: PropTypes.object.isRequired
+}
+
+/**
+ * DataTypeMapLines
+ * Renders a DataTypeMap on multiple lines
+ */
+
+export const DataTypeMapLines = ({
+  label,
+  depth,
+  dataTypeMap
+}: DataTypeMapLinesProps) => (
+  <>
+    {label && <Line label={label} depth={depth} />}
+    {Object.entries(dataTypeMap).map(([key, dt]) => (
+      <DataTypeLines
+        key={key}
+        label={key}
+        depth={label ? (depth ?? 0) + 1 : depth}
```

Comment:
Think we could simplify things a bit if we give `depth` a `defaultProps` value of `1` - since (i think?), the only time you'd need to include a >1 `depth` value is recursively... the developer him/herself probably doesn't need to pass depth unless in a special case that we don't yet have?